src/AlgebraicGeometry/Schemes/PrincipalOpenSubset/Objects/Attributes.jl

```diff
@@ -16,8 +16,11 @@ ambient_scheme(U::PrincipalOpenSubset) = U.X
 complement_equation(U::PrincipalOpenSubset) = U.f::elem_type(OO(ambient_scheme(U)))
 
 ### assure compatibility with SpecOpen 
-gens(U::PrincipalOpenSubset) = [lifted_numerator(complement_equation(U))]
+complement_equations(U::PrincipalOpenSubset) = [lifted_numerator(complement_equation(U))]
+number_of_complement_equations(U::PrincipalOpenSubset) = 1
 ngens(U::PrincipalOpenSubset) = 1
+gens(U::PrincipalOpenSubset) = [U]
+gen(U::PrincipalOpenSubset, i::Int) = (i == 1 ? U : error("index out of range"))
```

Comment:
This is not an objection or change request, just a genuine question: why even define `gen`, `gens` and `ngens` here? Do you expect to write "generic" code which will accept `PrincipalOpenSubset` but also other types that don't define `complement_equations`, and so that generic code will somehow use `gens` ?

Reply:
As far as I remember we basically consider `PrincipalOpenSubset` and `SpecOpen` as covered schemes implicitly. 
One way to access the affine charts is via some index.
Aside: I wonder if we want to rename `SpecOpen` into `ZariskiOpen` after all it is not (even isomorphic to) the spectrum of a ring.

Reply:
Shouldn't this be again an instance of complement_equation instead of gens? It seems to have changed in specOpen, but not in principalOpenSubset

Reply:
The reason is that `SpecOpen` and `PrincipalOpenSubset` are treated in parallel in glueings. In some places I wanted to write generic code which wouldn't distinguish between those two cases. And that's why I thought this was a good idea: It makes the `PrincipalOpenSubset`s implement the `SpecOpen` interface.

Reply:
The reason is that `SpecOpen` and `PrincipalOpenSubset` are treated in parallel in glueings. In some places I wanted to write generic code which wouldn't distinguish between those two cases. And that's why I thought this was a good idea: It makes the `PrincipalOpenSubset`s implement the `SpecOpen` interface.